### PR TITLE
Refactor nxsunit

### LINKED ISSUE #5: The celsius/kelvin conversion in nxsunit is not correct.
Celsius and kelvin temperature units are currently considered equivalent, with no offset between them.

A related implementation from another project is attached, with support for celsius, kelvin and Fahrenheit.

[nxsunit.txt](https://github.com/SasView/sasview/files/4714202/nxsunit.txt)

### PR DESCRIPTION
**Pull request by krzywon**
This overhauls the nxsunit module to allow scaling with offsets, and scaling of multi-dimensional and complex units, e.g. SESANS A^-2*cm^-1 will treat the A^-2 and cm^-1 as separate components, each independently scalable. Unit tests were moved from within the module to the global unit test suite.

Fixes #5
Fixes(?) sasview/sasview#2035 - @pkienzle what do you think?

### CODE REVIEW

**Comment by butlerpd** [comment]
Still waiting for review by @pkienzle and @lucas-wilkins (though the latter will be after Jan 1 I guess?). @wpotrzebowski may also take a look.

File: sasdata/data_util/nxsunit.py
Comment (wpotrzebowski) [inline sasdata/data_util/nxsunit.py:294]: Is Hertz/Hz also allowed?

File: sasdata/data_util/nxsunit.py
Comment (wpotrzebowski) [inline sasdata/data_util/nxsunit.py:397]: Maybe better to return scalebase and scaleoffset as two separate variables?

**Review by wpotrzebowski — COMMENTED** [review]
I am not that familiar with this code base but code looks good. Minor stylistic remark There are few places with for loops with single letter variables. I haven't tested functionality yet.

File: sasdata/data_util/nxsunit.py
Comment (lucas-wilkins) [reply sasdata/data_util/nxsunit.py:294]: think they need the ignorecase flags set

**Review by lucas-wilkins — APPROVED** [review]
Looks OK

File: sasdata/data_util/nxsunit.py
Comment (krzywon) [reply sasdata/data_util/nxsunit.py:98]: I agree, but this class works mostly in SI units. Here is a list of edge cases we might come up against where this fails, based on my understanding of [the NIST guide to SI](https://www.nist.gov/pml/special-publication-811/nist-guide-si-chapter-9-rules-and-style-conventions-spelling-unit-names):

- Anything Imperial that isn't a temperature
  - Potential solution: add imperial lengths and weights
  - What I am willing to do for this PR: I will continue to omit this. The scattering community doesn't work in inch(es)/feet/pounds/etc.
- `microohm` and `kiloohm` will be generated by the class, but `microhm` and `kilohm` are the correct forms.
  - Potential solution: explicitly add these as resistance units
  - What I am willing to do for this PR: I could implement this pretty quickly if there is a call for it.
- units in long-form, e.g. `meters per second squared` -or- `meters per square second`
  - Potential solution 1: convert `<square|cubic> <unit>` or `<unit> <squared|cubed>` to `unit^<2|3>`
  - Potential solution 2: convert `per` to `/`
  - What I am willing to do for this PR: I ignored this because long-form is typically reserved for textual content, not data units, but implementation would be fairly quick. I'd also need to add some reasonable unit tests.

**Comment by butlerpd** [comment]
is this ready to merge?

**Pull request merged**
